model_analyzer/config/generate/perf_analyzer_config_generator.py

```diff
@@ -154,87 +152,108 @@ def set_last_results(
             measurement = [max(valid_measurements)]
 
             self._last_results = measurement
-            self._concurrency_results.extend(measurement)
+            self._parameter_results.extend(measurement)
 
-    def _create_concurrency_list(self, cli_config: ConfigCommandProfile,
-                                 model_parameters: dict) -> List[int]:
+    def _create_parameter_list(self, model_parameters: dict) -> List[int]:
```

Comment:
I'm concerned about this code. This if/else chain and the decision of rr vs conc doesn't seem to be captured in a function or variable, so other parts of the code could make different decisions.  It seems possible that we are in request-rate mode and would return a list of concurrencies, or vice versa?

For example, what if request_rate is specified in the yaml? It seems we would hit the case on line 163 here, but wants_request_rate would still return false? So line 203 would use those values but tag them as concurrency instead? Am I misreading it? Is there unit testing for the yaml specification case?

Reply:
My `_config_specifies_request_rate()` method handles both CLI and yaml. I also have unit testing that checks for request_rate as a model parameter.

Reply:
I've refactored this

Reply:
ty that is what i had envisioned 👍